storage/spanstore/downsampling_writer.go

```diff
@@ -100,7 +100,8 @@ func (ds *DownsamplingWriter) WriteSpan(span *model.Span) error {
 	return ds.spanWriter.WriteSpan(span)
 }
 
-func (ds *DownsamplingWriter) shouldDownsample(span *model.Span) bool {
+// ShouldDownsample returns if the span should be sampled or not
+func (ds *DownsamplingWriter) ShouldDownsample(span *model.Span) bool {
```

Comment:
IMHO, a better way to go about this would be to refactor `ShouldDownsample` into a type that solely provides this functionality. Exposing this here forces an user of this API to create a writer when all they want is something that tells them whether a trace should be downsampled.

Reply:
+1

Reply:
Will  refactor.